Sources/web3swift/Tokens/ERC1633/Web3+ERC1633.swift

```diff
@@ -151,32 +132,41 @@ public class ERC1633: IERC1633, ERC20BaseProperties {
             throw Web3Error.inputError(desc: "Can not parse inputted amount")
         }
 
-        let tx = contract.createWriteOperation("approve", parameters: [spender, value] as [AnyObject] )!
+        let tx = contract.createWriteOperation("approve", parameters: [spender, value] as [AnyObject])!
         return tx
     }
 
     func parentToken() async throws -> EthereumAddress {
-        let contract = self.contract
-        self.transaction.callOnBlock = .latest
-        let result = try await contract.createReadOperation("parentToken", parameters: [] as [AnyObject], extraData: Data() )!.callContractMethod()
+        transaction.callOnBlock = .latest
```

Comment:
All these `transaction.callOnBlock = .latest` make no difference as the transaction is not set to `contract.transaction`.
It didn't make any difference all the time before 🤯
It should be safe to call `updateTransactionAndContract(from: address)`.

All these changes in the PR are a step towards good code but perfectly there should be no `transaction` and `contract.transaction` variables at all. But it's out of scope right now.

Reply:
haha ok, will remove